src/cli/main.c

```diff
@@ -88,35 +88,69 @@ parse_profile_options(struct cli_cmdline *cmdline,
 
 static errno_t activate(struct cli_cmdline *cmdline)
 {
+    struct authselect_profile *profile = NULL;
+    const char **features = NULL;
     const char *profile_id;
-    const char **features;
+    const char *requirements;
     int enforce = 0;
+    int quiet = 0;
     errno_t ret;
 
     struct poptOption options[] = {
         {"force", 'f', POPT_ARG_VAL, &enforce, 1, _("Enforce changes"), NULL },
+        {"quiet", 'q', POPT_ARG_VAL, &quiet, 1, _("Do not print profile requirements"), NULL },
         POPT_TABLEEND
     };
 
     ret = parse_profile_options(cmdline, options, &profile_id, &features);
     if (ret != EOK) {
-        return ret;
+        goto done;
+    }
+
+    ret = authselect_profile(profile_id, &profile);
+    if (ret != EOK) {
+        ERROR("Unable to get profile information [%d]: %s",
+              ret, strerror(ret));
+        ret = ENOMEM;
+        goto done;
+    }
+
+    requirements = authselect_profile_requirements(profile, features);
```

Comment:
It looks like you don't free requirements anywhere.

Reply:
Ah, I see why. I didn't add it to `authselect_profile_free`